codec/src/main/java/io/netty/handler/codec/Headers.java

```diff
@@ -19,7 +19,14 @@
 import java.util.Map.Entry;
 import java.util.Set;
 
-public interface Headers<T> extends Iterable<Entry<T, T>> {
+/**
+ * Common interface for {@link Headers} which represents a mapping of key to value.
+ * Duplicate keys may be allowed by implementations.
+ *
+ * @param <K> the type of the header name.
+ * @param <V> the type of the header value.
+ */
+public interface Headers<K, V> extends Iterable<Entry<K, V>> {
```

Comment:
All types are now `Headers<CharSequence, CharSequence>`. However I'm not sure if it is necessary or if there would be significant gains to limit the generic nature of this class. If anything could we handle this in a followup PR?